substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixRawFileOperationSupport.java

```diff
@@ -0,0 +1,156 @@
+    @Override
+    public SignedWord size(RawFileDescriptor fd) {
+        int posixFd = getPosixFileDescriptor(fd);
+        return PosixStat.getSize(posixFd);
+    }
+
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    @Override
+    public SignedWord position(RawFileDescriptor fd) {
+        int posixFd = getPosixFileDescriptor(fd);
+        return Unistd.NoTransitions.lseek(posixFd, WordFactory.signed(0), Unistd.SEEK_CUR());
+    }
+
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    @Override
+    public boolean seek(RawFileDescriptor fd, SignedWord position) {
+        int posixFd = getPosixFileDescriptor(fd);
+        SignedWord newPos = Unistd.NoTransitions.lseek(posixFd, position, Unistd.SEEK_SET());
+        return position.equal(newPos);
+    }
+
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    @Override
+    public boolean write(RawFileDescriptor fd, Pointer data, UnsignedWord size) {
+        int posixFd = getPosixFileDescriptor(fd);
+
+        Pointer position = data;
+        UnsignedWord remaining = size;
+        while (remaining.aboveThan(0)) {
+            SignedWord writtenBytes;
+            do {
+                writtenBytes = Unistd.NoTransitions.write(posixFd, position, remaining);
+            } while (writtenBytes.equal(-1) && CErrorNumber.getCErrorNumber() == Errno.EINTR());
+
+            if (writtenBytes.equal(-1)) {
+                return false;
+            }
+            position = position.add((UnsignedWord) writtenBytes);
+            remaining = remaining.subtract((UnsignedWord) writtenBytes);
+        }
+        return true;
+    }
+
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    @Override
+    public SignedWord read(RawFileDescriptor fd, Pointer buffer, UnsignedWord bufferSize) {
+        int posixFd = getPosixFileDescriptor(fd);
+
+        SignedWord readBytes;
+        do {
+            readBytes = Unistd.NoTransitions.read(posixFd, buffer, bufferSize);
+        } while (readBytes.equal(-1) && CErrorNumber.getCErrorNumber() == Errno.EINTR());
+
+        return readBytes;
+    }
+
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    private static int getPosixFileDescriptor(RawFileDescriptor fd) {
+        int result = (int) fd.rawValue();
+        assert result == fd.rawValue();
+        return result;
+    }
+
+    private static int parseMode(FileAccessMode mode) {
+        if (mode == FileAccessMode.READ) {
+            return Fcntl.O_RDONLY();
+        } else if (mode == FileAccessMode.READ_WRITE) {
+            return Fcntl.O_RDWR() | Fcntl.O_CREAT();
+        } else if (mode == FileAccessMode.WRITE) {
+            return Fcntl.O_WRONLY() | Fcntl.O_CREAT();
+        }
+
+        throw new IllegalArgumentException("Illegal file access mode '" + Objects.toString(mode) + "'.");
```

Comment:
I don't think that you need Objects.toString() here, and can simply use 'mode'. Unless you want it formatted as octal number, but then you need more than Objects.toString() ;-)